src/services/staff/staff-router.ts

```diff
@@ -74,4 +77,34 @@ staffRouter.post("/attendance/", strongJwtVerification, async (req: Request, res
     return res.status(StatusCode.SuccessOK).send({ status: "Success" });
 });
 
+staffRouter.get("/shift/", strongJwtVerification, async (_: Request, res: Response, next: NextFunction) => {
+    const payload: JwtPayload | undefined = res.locals.payload as JwtPayload;
+
+    if (!hasStaffPerms(payload)) {
+        return next(new RouterError(StatusCode.ClientErrorForbidden, "Forbidden"));
+    }
+
+    try {
```

Comment:
No need for try-catches now :)

Reply:
Does the code automatically recover if theres an error thrown? or is there some other handling which needs to be added?

Reply:
yeah - eg a database call for type X becomes:

`const data: X | null = Models...`, and it's null if nothing is returned - basically it doesn't throw an error, but is a wrapper that returns null for no data found